### PR TITLE
feat(ai-service-dedupe-e2e): added e2e test for scan notification

### PR DESCRIPTION
**Pull request by waabid**
#### Description of changes

Adds an e2e test for a successful scan notification (that should pass).

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/web-workers/src/orchestration-steps.ts [hunk 1]
```diff
@@ -147,19 +155,88 @@ export class OrchestrationStepsImpl implements OrchestrationSteps {
...
+        const waitEndTime = waitStartTime
+            .clone()
+            .add(this.availabilityTestConfig.maxScanCompletionNotificationWaitTimeInSeconds, 'seconds');
+        const scanWaitIntervalInSeconds = this.availabilityTestConfig.scanWaitIntervalInSeconds;
+        let scanStatusResponse: SerializableResponse;
+
+        this.logOrchestrationStep('Starting wait for scan completion notification');
+
+        while (
+            completedNotificationStates.indexOf(notificationState) === -1 &&
```
Comment (pownkel) [inline packages/web-workers/src/orchestration-steps.ts:186]: Nit
```suggestion
            completedNotificationStates.includes(notificationState) &&
```

File: packages/web-workers/src/orchestration-steps.ts [hunk 2]
```diff
@@ -147,19 +155,88 @@ export class OrchestrationStepsImpl implements OrchestrationSteps {
...
+
+            scanStatusResponse = yield* this.callGetScanStatusActivity(scanId);
+            scanStatus = yield* this.getScanStatus(scanStatusResponse);
+
+            notificationState = scanStatus.notification.state;
+        }
+
+        const totalWaitTimeInSeconds = moment.utc(this.context.df.currentUtcDateTime).diff(moment.utc(waitStartTime), 'seconds');
+
+        if (completedNotificationStates.indexOf(notificationState) !== -1) {
```
Comment (pownkel) [inline packages/web-workers/src/orchestration-steps.ts:212]: Nit
```suggestion
        if (completedNotificationStates.includes(notificationState)) {
```

**Review by pownkel — APPROVED** [review]
Approved with a few minor comments

**Comment by waabid** [comment]
Will address nits in a future PR.

**Pull request merged**